public/index.html

```diff
@@ -474,46 +535,29 @@
           navigator.clipboard.writeText(dateInput.value);
           dateInput.style.fontWeight = "bold";
           dateInputMessage.textContent = "Copied to clipboard";
+          setTimeout(() => {
+            dateInput.style.fontWeight = "normal";
+            dateInputMessage.textContent = "Click to copy";
+          }, 1000);
         });
 
-        document.getElementById("englishDate").textContent = formatDateTime(now);
-      };
+        // Update English date
+        document.getElementById("englishDate").textContent =
+          `${nepalYear}-${nepalMonth.toString().padStart(2, "0")}-${nepalDay
+            .toString()
+            .padStart(2, "0")} ` + `${nepalHour}:${nepalMinute}:${nepalSecond}`;
 
-      const fetchServerTime = async () => {
-        try {
-          const response = await fetch(
-            `https://yarsa.io/apis/utils/nepal-time.php`,
-            {
-              cf: {
-                timeoutMs: 2000,
-              },
-            }
-          );
-
-          if (response.ok) {
-            serverTime = new Date(await response.text());
-            startTime = performance.now(); // Record the start time for precise updates
-          }
-        } catch (error) {
-          console.error(`Failed to fetch time from server:`, error);
-          // Retry fetching the server time after a short delay
-          setTimeout(fetchServerTime, 1000);
-        }
-      };
-
-      const formatDateTime = (date) => {
-        const year = date.getFullYear();
-        const month = padNumberWithZeroToMakeTwoDigits(date.getMonth() + 1); // Months are 0-indexed
-        const day = padNumberWithZeroToMakeTwoDigits(date.getDate());
-        const hours = padNumberWithZeroToMakeTwoDigits(date.getHours());
-        const minutes = padNumberWithZeroToMakeTwoDigits(date.getMinutes());
-        const seconds = padNumberWithZeroToMakeTwoDigits(date.getSeconds());
-
-        return `${year}-${month}-${day} ${hours}:${minutes}:${seconds}`;
+        // Update copy field
+        document.getElementById("rawDateTime").value = `${bsYear}-${bsMonth
+          .toString()
+          .padStart(2, "0")}-${bsDate.toString().padStart(2, "0")}`;
       };
 
+      let startTime;
       document.addEventListener("DOMContentLoaded", async () => {
         await fetchServerTime();
+        startTime = Date.now();
+        updateDateTime();
```

Comment:
You'll have to move this line before the await in line 558 if you want to solve #2. Other parts of the PR look good. I verified that #4 is indeed fixed.

```
      document.addEventListener("DOMContentLoaded", async () => {
+       updateDateTime();
        await fetchServerTime();
        startTime = Date.now();
-       updateDateTime();
        setInterval(updateDateTime, 1000);
      });